python/simulation.py

```diff
@@ -100,13 +100,27 @@ def py_v3_to_vec(dims, iterable, is_cylindrical=False):
         raise ValueError("Invalid dimensions in Volume: {}".format(dims))
 
 class DiffractedPlanewave(object):
+    """
+    For mode decomposition, specify a diffracted planewave in homogeneous media. Should be passed as the `bands` argument of `get_eigenmode_coefficients` or `band_num` of `get_eigenmode`.
+    """
     def __init__(self,
                  g=None,
                  axis=None,
                  s=None,
                  p=None):
+        """
+        Construct a `DiffractedPlanewave`.
+
+        + **`g` [ list of 3 `integer`s ]** — The diffraction order $(m_x,m_y,m_z)$ corresponding to the wavevector $(k_x+2\pi m_x/\Lambda_x,k_y+2\pi m_y/\Lambda_y,k_z+2\pi m_z/\Lambda_z)$. Elements should be non-zero only in the $d$-1 periodic directions of a $d$ dimensional cell (e.g., a plane in 3d) in which the mode monitor extends the entire length of the cell.
+
+        + **`axis` [ `Vector3` ]** — The reference axis used in combination with the mode's wavevector (via the cross product) to define the plane of incidence. If `None`, defaults to the first direction that lies in the plane of the monitor (e.g., $y$ direction for a $yz$ plane in 3d, either $x$ or $y$ in 2d).
```

Comment:
The plane of incidence for each planewave (used to define the S and P polarizations below) is defined to be the plane that contains the `axis` vector and the planewave's wavevector.

Reply:
The definition of the `axis` vector has been updated.